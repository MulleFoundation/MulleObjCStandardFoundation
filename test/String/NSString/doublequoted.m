#import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>


@implementation NSString( SingleQuoted)

- (NSString *) mulleSingleQuoteEscapedString
{
   NSString  *s;

   s = [self stringByReplacingOccurrencesOfString:@"'"
                                       withString:@"'''"];
   return( s);
}

@end



static void   test( NSString *cmdline)
{
   NSArray       *array;
   NSString      *s;
   unsigned int  i;

   array = [cmdline mulleComponentsSeparatedByWhitespaceWithDoubleQuoting];

   mulle_printf( "'%@':\n", [cmdline mulleSingleQuoteEscapedString]);

   i = 0;
   for( s in array)
      mulle_printf( "   #%d: '%@'\n", i++, [s mulleSingleQuoteEscapedString]);

   mulle_printf( "\n");
}


int  main( int argc, char *argv[])
{
   test( @"\"VfL\"\" Bochum\"\" 1848\""); // one string

   test( @"VfL Bochum 1848");
   test( @"\"\\\"");
   test( @"\\");  // broken stuff

   test( @"\"a\\\"\\x\\\\b\"");
   test( @"\"a\\\"b\"");
   test( @"\"a\\\"\"");

   test( @"\" a \" \" b \"");

   test( @"  \"a\"  \"b\"  ");
   test( @"\"a\"  \"b\"  ");
   test( @"  \"a\"  \"b\"");
   test( @"\"a\"  \"b\"");
   test( @"\"a\" \"b\"  ");
   test( @"  \"a\" \"b\"");

   test( @" \"a\" \"b\" ");
   test( @"\"a\" \"b\" ");
   test( @" \"a\" \"b\"");

   test( @"\"a\" \"b\"");
   test( @"\"a\"");
   test( @"");
   test( @"\"\"");

   test( @"  a  b  ");
   test( @"a  b  ");
   test( @"  a  b");
   test( @"a  b");
   test( @"a b  ");
   test( @"  a b");

   test( @" a b ");
   test( @"a b ");
   test( @" a b");

   test( @"a b");
   test( @"a");
   test( @"");

   return( 0);
}
