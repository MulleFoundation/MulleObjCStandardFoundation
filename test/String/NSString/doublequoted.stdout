'"VfL"" Bochum"" 1848"':
   #0: 'VfL Bochum 1848'

'VfL Bochum 1848':
   #0: 'VfL'
   #1: 'Bochum'
   #2: '1848'

'"\"':
   #0: ''

'\':
   #0: '\'

'"a\"\x\\b"':
   #0: 'a"\x\b'

'"a\"b"':
   #0: 'a"b'

'"a\""':
   #0: 'a"'

'" a " " b "':
   #0: ' a '
   #1: ' b '

'  "a"  "b"  ':
   #0: 'a'
   #1: 'b'

'"a"  "b"  ':
   #0: 'a'
   #1: 'b'

'  "a"  "b"':
   #0: 'a'
   #1: 'b'

'"a"  "b"':
   #0: 'a'
   #1: 'b'

'"a" "b"  ':
   #0: 'a'
   #1: 'b'

'  "a" "b"':
   #0: 'a'
   #1: 'b'

' "a" "b" ':
   #0: 'a'
   #1: 'b'

'"a" "b" ':
   #0: 'a'
   #1: 'b'

' "a" "b"':
   #0: 'a'
   #1: 'b'

'"a" "b"':
   #0: 'a'
   #1: 'b'

'"a"':
   #0: 'a'

'':
   #0: ''

'""':
   #0: ''

'  a  b  ':
   #0: 'a'
   #1: 'b'

'a  b  ':
   #0: 'a'
   #1: 'b'

'  a  b':
   #0: 'a'
   #1: 'b'

'a  b':
   #0: 'a'
   #1: 'b'

'a b  ':
   #0: 'a'
   #1: 'b'

'  a b':
   #0: 'a'
   #1: 'b'

' a b ':
   #0: 'a'
   #1: 'b'

'a b ':
   #0: 'a'
   #1: 'b'

' a b':
   #0: 'a'
   #1: 'b'

'a b':
   #0: 'a'
   #1: 'b'

'a':
   #0: 'a'

'':
   #0: ''

